src/java.base/share/classes/sun/security/tools/keytool/Main.java

```diff
@@ -4810,6 +5026,75 @@ private void checkWeak(String label, CRL crl, Key key) {
         }
     }
 
+    private void buildTrustedCerts() {
+        try {
+            KeyStore caks = KeyStoreUtil.getCacertsKeyStore();
+            if (caks != null) {
+                Enumeration<String> aliases = caks.aliases();
+                while (aliases.hasMoreElements()) {
+                    String a = aliases.nextElement();
+                    try {
+                        trustedCerts.add((X509Certificate)caks.getCertificate(a));
+                    } catch (Exception e2) {
+                        // ignore, when a SecretkeyEntry does not include a cert
+                    }
+                }
+            }
+        } catch (Exception e) {
+            // Ignore, if cacerts cannot be loaded
+        }
+    }
+
+    private TrustAnchor findTrustAnchor(List<X509Certificate> chain) {
```

Comment:
I would consider having an initial check that returns `null` if `chain.isEmpty()`. Not sure if that is a valid scenario, but it would avoid an `IndexOOBException` just in case.

Reply:
Added the initial check.